portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,35 +34,42 @@
 public class DataServlet extends HttpServlet {
 
   private ArrayList<String> messages = new ArrayList<String>();
+  private ArrayList<String> firstName = new ArrayList<String>();
+  private ArrayList<String> lastName = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
     response.setContentType("application/json;");
-    
+
     Query query = new Query("Messages").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-
     PreparedQuery results = datastore.prepare(query);
 
+    String numResultsString = (request.getQueryString());
+    int numResults;
```

Comment:
Since the default numResults is 1, you can simplify the code like below. There are two benefits of the following implementation: 1) the default value is clear to be set to one; 2) If we add more options (say "num-results=1000") in the future, we don't need to change the code.

int numResults = 1;
String numResultsStringPrefix = "num-results=";
if (numResultsString != null && numResultsString.startWith(numResultsStringPrefix) {
    // I leave this part for you to finish. Can you parse the numResults from the numResultsString? 
}

Reply:
You can also use `request.getParameter("num-results")` instead of getting query string and let library do the parsing.

https://docs.oracle.com/javaee/6/api/javax/servlet/ServletRequest.html#getParameter(java.lang.String)

Reply:
Then you can cast it to `Integer.ParseInt(<string>)`, just keep in mind there can be bad inputs.

Reply:
Oh so that way, I wouldn't need to do the substring where the number is, I can just parse it. About the bad inputs, I have it as a dropdown and the client selects the number of comments to be shown, so I am having trouble thinking of bad inputs for this case. Can I get an example?